src/assets/ManifestAsset.js

```diff
@@ -0,0 +1,168 @@
+const path = require('path');
+const Asset = require('../Asset');
+
+/**
+ * A shared asset that handles:
+ * - PWA .webmanifest
+ * - PWA manifest.json
+ * - WebExtension manifest.json
+ */
+class ManifestAsset extends Asset {
+  constructor(name, options) {
+    super(name, options);
+    this.type =
+      path.basename(name) === 'manifest.json' ? 'json' : 'webmanifest';
```

Comment:
Consider extracting to separate method, e.g. `this.type = this.typeFor(name)`

We may want to test the file beyond just it's `basename` to determine the `type` in future, for example `manifest.js` or `manifest.json5` that gets transformed to `manifest.json` on the way out.